src/syscheckd/db/fim_db.c

```diff
@@ -422,17 +422,25 @@ int fim_db_finalize_stmt(fdb_t *fim_sql) {
 
 void fim_db_check_transaction(fdb_t *fim_sql) {
     time_t now = time(NULL);
+    int query_result = FIMDB_OK;
 
     if (fim_sql->transaction.last_commit + fim_sql->transaction.interval <= now) {
         if (!fim_sql->transaction.last_commit) {
             fim_sql->transaction.last_commit = now;
             return;
         }
 
-        // If the completion of the transaction fails, we do not update the timestamp
-        if (fim_db_exec_simple_wquery(fim_sql, "END;") != FIMDB_ERR) {
-            mdebug1("Database transaction completed.");
-            fim_sql->transaction.last_commit = now;
+        if (!sqlite3_get_autocommit(fim_sql->db)) {
+            // A transaction has been initiated by a BEGIN command and it's in progress
+            if (query_result = fim_db_exec_simple_wquery(fim_sql, "END;"), query_result != FIMDB_ERR) {
+                // Updating timestamp only after a successful transaction end
+                mdebug1("Database transaction completed.");
+                fim_sql->transaction.last_commit = now;
+            }
```

Comment:
We can invert this if condition, returning directly if it fails. This should allow us to remove the `query_result` auxiliar variable and got straight to executing a `BEGIN` if it works as expected. IMO, it would be easier to read the code this way.

```suggestion
            if (fim_db_exec_simple_wquery(fim_sql, "END;") == FIMDB_ERR) {
                mdebug1("Failed executing 'END' operation");
                return;
            }
            // Updating timestamp only after a successful transaction end
            mdebug1("Database transaction completed.");
            fim_sql->transaction.last_commit = now;
```

Reply:
You are right, now it's more clear.
The DEBUG message was omitted because `fim_db_exec_simple_wquery` already prints an error message.